community/kernel/src/test/java/org/neo4j/kernel/impl/locking/StopCompatibility.java

```diff
@@ -556,16 +554,9 @@ private void assertNoLocksHeld()
 
     private void assertThreadIsWaitingForLock( LockAcquisition lockAcquisition ) throws Exception
     {
-        for ( int i = 0; i < 30; i++ )
+        for ( int i = 0; i < 30 && !suite.isAwaitingLockAcquisition( lockAcquisition.executor.waitUntilWaiting() ); i++ )
```

Comment:
It is equivalent and even asserts more than we did previously. In this piece of code we wait for the thread to go into a _waiting_ state AND see that it waits in the correct place. The previous code only verified that the thread hadn't acquired the lock in N seconds, it could have been caused by anything.